google-oauth-client/src/test/java/com/google/api/client/auth/oauth2/TokenRequestTest.java

```diff
@@ -45,4 +50,36 @@ static void check(TokenRequest request, String grantType) {
     assertEquals(JSON_FACTORY, request.getJsonFactory());
     assertEquals(AUTHORIZATION_SERVER_URL, request.getTokenServerUrl());
   }
+
+  public void testScopes() throws IOException {
+    TokenRequest request =
+        new TokenRequest(TRANSPORT, JSON_FACTORY, AUTHORIZATION_SERVER_URL, "foo")
+            .setScopes(ImmutableList.of("scope1"));
+    HttpContent content = new UrlEncodedContent(request);
+    ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
+    content.writeTo(outputStream);
+    String encoded = new String(outputStream.toByteArray());
```

Comment:
need a character set here and below for robustness across platforms and locales

Reply:
Done.